libraries/AP_HAL_Linux/HAL_Linux_Class.cpp

```diff
@@ -163,6 +163,8 @@ static RCInput_SoloLink rcinDriver;
 static RCInput_Navio2 rcinDriver;
 #elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_RST_ZYNQ
 static RCInput_RCProtocol rcinDriver{"/dev/ttyPS0", NULL};
+#elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_VNAV
+static RCInput_RCProtocol rcinDriver{"/dev/null", NULL};
```

Comment:
Is this a leftover or just a placeholder?

Reply:
it is needed to allow for rcinput via SERIALn_PROTOCOL=23 (for rcin)
ie. we need a AP_RCProtocol object, and in HAL_Linux this is how we get it

Reply:
I've added a comment

Reply:
I knew this was for SERIALn_PROTOCOL. What I found strange was the use of both NULL and /dev/null (we should use either one or the other). Also neither one or the other will inject input in AP::RC().  I guess `ioctl(fd, TCSETS2, &tio)` is also failing when you try that ioctl on /dev/null and the file descriptor is closed.

Where are you injecting RC input from?  Maybe AP_RCProtocol is not the right one for this board?

Reply:
I knew this was for SERIALn_PROTOCOL. What I found strange was the use of both NULL and /dev/null (we should use either one or the other). Also neither one or the other will inject input in AP::RC().  I guess `ioctl(fd, TCSETS2, &tio)` is also failing when you try that ioctl on /dev/null and the file descriptor is closed.

Where are you injecting RC input from?  Maybe AP_RCProtocol is not the right one for this board?